commit 6fd451ce607701a575037ef260fc70061e16b403
Author: Randy Dunlap <rdunlap@infradead.org>
Date:   Tue Jan 24 15:18:49 2017 -0800

    mn10300: fix build error of missing fpu_save()
    
    
    [ Upstream commit 3705ccfdd1e8b539225ce20e3925a945cc788d67 ]
    
    When CONFIG_FPU is not enabled on arch/mn10300, <asm/switch_to.h> causes
    a build error with a call to fpu_save():
    
      kernel/built-in.o: In function `.L410':
      core.c:(.sched.text+0x28a): undefined reference to `fpu_save'
    
    Fix this by including <asm/fpu.h> in <asm/switch_to.h> so that an empty
    static inline fpu_save() is defined.
    
    Link: http://lkml.kernel.org/r/dc421c4f-4842-4429-1b99-92865c2f24b6@infradead.org
    Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Reviewed-by: David Howells <dhowells@redhat.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/mn10300/include/asm/switch_to.h b/arch/mn10300/include/asm/switch_to.h
index 393d311..67e333a 100644
--- a/arch/mn10300/include/asm/switch_to.h
+++ b/arch/mn10300/include/asm/switch_to.h
@@ -16,7 +16,7 @@
 struct task_struct;
 struct thread_struct;
 
-#if !defined(CONFIG_LAZY_SAVE_FPU)
+#if defined(CONFIG_FPU) && !defined(CONFIG_LAZY_SAVE_FPU)
 struct fpu_state_struct;
 extern asmlinkage void fpu_save(struct fpu_state_struct *);
 #define switch_fpu(prev, next)						\
